src/main/java/net/snowflake/ingest/connection/TelemetryService.java

```diff
@@ -122,6 +127,26 @@ public void reportCpuMemoryUsage(Histogram cpuUsage) {
     }
   }
 
+  public void reportBatchOffsetMismatch(
+      String channelName,
+      String prevBatchEndOffset,
+      String startOffset,
+      String endOffset,
+      long rowCount) {
+    if (rateLimiter.tryAcquire()) {
```

Comment:
also this is a blocking call, should we do this async or may be use 
tryAcquire(long timeout,
                  unit)
with a timeout?

Reply:
Not sure if I understand, but tryAcquire() means the timeout is 0 and it will immediately return if it can't get it which is better than adding a timeout?

Reply:
got it, yeah this is better than adding a timeout.